src/components/shapes/helpers.js

```diff
@@ -208,23 +231,22 @@ exports.getPathString = function(gd, options) {
         if(ya && ya.type === 'date') y2p = exports.decodeDate(y2p);
         return convertPath(options, x2p, y2p);
     }
-
     if(options.xsizemode === 'pixel') {
         var xAnchorPos = x2p(options.xanchor);
-        x0 = xAnchorPos + options.x0;
-        x1 = xAnchorPos + options.x1;
+        x0 = xAnchorPos + options.x0 + xShiftStart;
```

Comment:
It would be nice if you could declare a centralize function to handle these adjustments here and in other files if used.

Considering my comment about possibility of snapping to the (category) grid [here](https://github.com/plotly/plotly.js/pull/7005/files#r1633371678), you may need to round the current value before adding start and end shifts. Having a centralized function would help testing that idea later.

Thank you!

Reply:
I moved the logic for calculating the shift in pixels into a centralized function, I did not yet include any rounding.